### PR TITLE
Add support for Rails 6 beta

### PR DESCRIPTION
**Pull request by ionut998**
Allow newer versions of actionpack and railties

### CODE REVIEW

**Comment by javierjulio** [comment]
@ionut998 thanks!

@deivid-rodriguez I think it's a good time to set up a Rails 6 beta CI build, what do you think?

**Comment by ionut998** [comment]
@javierjulio I just added the setup for Rails 6 beta CI build

File: test/gemfiles/Gemfile-Rails-6-0-beta [hunk 1]
```diff
@@ -0,0 +1,10 @@
+source 'https://rubygems.org'
+
+gemspec :path => '../..'
+
+gem 'rails', '6.0.0.beta1'
```
Comment (javierjulio) [inline test/gemfiles/Gemfile-Rails-6-0-beta]: Actually beta2 was just released a few days ago!

```suggestion
gem 'rails', '6.0.0.beta2'
```

**Review by javierjulio — COMMENTED** [review]
Thanks for contributing the CI build as well. 👍🏻

**Review by javierjulio — APPROVED** [review]
Awesome, thanks a lot @ionut998! This looks great. 👍🏻

File: .circleci/config.yml [hunk 2]
```diff
@@ -133,6 +133,16 @@ jobs:
...
+  ruby25rails60beta:
+    docker:
+    - image: circleci/ruby:2.5.1
+
+    environment:
+      BUNDLE_GEMFILE: test/gemfiles/Gemfile-Rails-6-0-beta
+
+    steps:
+      *test_steps
+
```
Comment (deivid-rodriguez) [inline .circleci/config.yml:145]: I guess we can add the whole matrix and test against all supported rubies since we are at it?

Comment (matheusvetor) [reply .circleci/config.yml:145]: We need also add ruby 2.6

**Review by matheusvetor — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (ionut998) [reply .circleci/config.yml:145]: @deivid-rodriguez according to the guideline Rails 6 requires ruby 2.5 or newer
https://edgeguides.rubyonrails.org/upgrading_ruby_on_rails.html#ruby-versions

Comment (deivid-rodriguez) [reply .circleci/config.yml:145]: You're absolutely right, my bad! This PR is going in then!

**Comment by deivid-rodriguez** [comment]
Thanks @ionut998!

**Comment by javierjulio** [comment]
Sorry I couldn't follow up @ionut998 but thanks a lot for this and the recent updates! Much appreciated. ❤️

**Comment by ionut998** [comment]
Thanks @javierjulio, Happy to help
Hope to see this in a new release

File: inherited_resources.gemspec [hunk 3]
```diff
@@ -20,7 +20,7 @@ Gem::Specification.new do |s|
   s.required_ruby_version = '>= 2.3'
 
   s.add_dependency("responders")
-  s.add_dependency("actionpack", ">= 5.0", "< 5.3")
-  s.add_dependency("railties", ">= 5.0", "< 5.3")
+  s.add_dependency("actionpack", ">= 5.0", "< 6.0")
+  s.add_dependency("railties", ">= 5.0", "< 6.0")
```
Comment (deivid-rodriguez) [inline inherited_resources.gemspec:24]: I was thinking about this a bit more, and I think this is a good strategy.

We allow any prereleases for now, but not the final version. We'll watch for compatibility from now on but hold off a new release until Rails 6.0 is released and we confirm we are compatible. Then bump the upper limit here to 6.1 for example and release a new version

**Pull request merged**